cppcache/integration-test-2/RegisterKeysTest.cpp

```diff
@@ -0,0 +1,199 @@
+  Cluster cluster{LocatorCount{1}, ServerCount{1}};
+  cluster.getGfsh()
+      .create()
+      .region()
+      .withName("region")
+      .withType("PARTITION")
+      .execute();
+
+  {
+    auto cache = createTestCache();
+    auto pf =
+        cache.getPoolManager().createFactory().setSubscriptionEnabled(true);
+    cluster.applyLocators(pf);
+    pf.create("default");
+    auto region = setupCachingProxyRegion(cache);
+
+    region->put("one", std::make_shared<CacheableInt16>(1));
+    region->put("two", std::make_shared<CacheableInt16>(2));
+    region->put("three", std::make_shared<CacheableInt16>(3));
+  }
+
+  {
+    auto cache2 = createTestCache();
+    auto pf =
+        cache2.getPoolManager().createFactory().setSubscriptionEnabled(true);
+    cluster.applyLocators(pf);
+    pf.create("default");
+    auto region2 = setupCachingProxyRegion(cache2);
+
+    auto&& entryBefore = region2->getEntry("one");
+    ASSERT_EQ(entryBefore, nullptr);
+
+    // 2nd parameter is getInitialValues, default is false, which leads to the
+    // first update notification being passed in with a NULL "oldValue".  Set it
+    // to true here, and verify the entry retrieved is valid, i.e. the initial
+    // value was retrieved.
+    region2->registerAllKeys(false, true);
+
+    auto&& uncastedEntry = region2->getEntry("one");
+    auto&& entryAfter =
+        std::dynamic_pointer_cast<CacheableInt16>(uncastedEntry->getValue());
+    ASSERT_NE(entryAfter, nullptr);
+    ASSERT_EQ(entryAfter->value(), 1);
+  }
+}
+
+/**
+ */
+TEST(RegisterKeysTest, RegisterAnyWithCachingRegion) {
+  Cluster cluster{LocatorCount{1}, ServerCount{1}};
+  cluster.getGfsh()
+      .create()
+      .region()
+      .withName("region")
+      .withType("PARTITION")
+      .execute();
+
+  {
+    auto cache = createTestCache();
+    auto pf =
+        cache.getPoolManager().createFactory().setSubscriptionEnabled(true);
+    cluster.applyLocators(pf);
+    pf.create("default");
+    auto region = setupCachingProxyRegion(cache);
+
+    region->put("one", std::make_shared<CacheableInt16>(1));
+    region->put("two", std::make_shared<CacheableInt16>(2));
+    region->put("three", std::make_shared<CacheableInt16>(3));
+
+    cache.close();
+  }
+
+  {
+    auto cache2 = createTestCache();
+    auto pf =
+        cache2.getPoolManager().createFactory().setSubscriptionEnabled(true);
+    cluster.applyLocators(pf);
+    pf.create("default");
+    auto region2 = setupCachingProxyRegion(cache2);
+    std::vector<std::shared_ptr<CacheableKey> > keys;
+    keys.push_back(std::make_shared<CacheableString>("one"));
+
+    auto&& entryBefore = region2->getEntry("one");
+    ASSERT_EQ(entryBefore, nullptr);
+
+    region2->registerKeys(keys, false, true);
+
+    auto&& entryAfterGet = std::dynamic_pointer_cast<CacheableInt16>(
+        region2->getEntry("one")->getValue());
+    ASSERT_NE(entryAfterGet, nullptr);
+    ASSERT_EQ(entryAfterGet->value(), 1);
+  }
+}
+
+/**
```

Comment:
Either add documentation to the test or don't but don't add empty documentation.